backend/flask_api.py

```diff
@@ -43,6 +50,85 @@ def index():
     """
     return make_response('', 404)
 
+# Create app
+app = Flask(__name__)
+app.config['DEBUG'] = True
+app.config['SECRET_KEY'] = 'super-secret'
+
+login_manager = LoginManager()
+login_manager.init_app(app)
+
+# MongoDB Config
+app.config['MONGODB_DB'] = 'knexDB'
+app.config['MONGODB_HOST'] = 'mongodb'
+app.config['MONGODB_PORT'] = 27017
+app.config['SECURITY_PASSWORD_HASH'] = 'pbkdf2_sha512'
+app.config['SECURITY_PASSWORD_SALT'] = 'THISISMYOWNSALT'
+# Create database connection object
+db = MongoEngine(app)
+
+
+class Role(db.Document, RoleMixin):
+    name = db.StringField(max_length=80, unique=True)
+    description = db.StringField(max_length=255)
+
+
+class User(db.Document, UserMixin):
+    email = db.StringField(max_length=255)
+    first_name = db.StringField(max_length=255)
+    last_name = db.StringField(max_length=255)
+    password = db.StringField(max_length=255)
+    active = db.BooleanField(default=True)
+    bio = db.StringField(max_length=255)
+    bookmarks = db.ListField(UUIDField(), default=[])
+    roles = db.ListField(db.ReferenceField(Role), default=[])
+
+
+class EmailConverter(BaseConverter):
+    regex = r"([a-z0-9!#$%&'*+\/=?^_`{|}~-]+(?:\." +\
+            "[a-z0-9!#$%&'*+\/=?^_`""{|}~-]+)" +\
+            "*(@|\sat\s)(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?" +\
+            "(\.|""\sdot\s))+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?)"
+
+
+# Setup Flask-Security
+user_datastore = MongoEngineUserDatastore(db, User, Role)
+security = Security(app, user_datastore)
+app.url_map.converters['email'] = EmailConverter
+
+
+@app.before_first_request
+def initialize_users():
+    user_role = user_datastore.find_or_create_role('user')
+    pw = encrypt_password("user")
```

Comment:
No it does not seem so...
If I remove the encrypt_password function, the password is stored as plaintext....(SECURITY_PASSWORD_HASH and SECURITY_PASSWORD_SALT are set...)